java/src/main/java/com/mlt/converter/mvt/MvtUtils.java

```diff
@@ -50,6 +61,23 @@ public static JtsMvt decodeMvt2Fast(ByteArrayInputStream mvtTile) throws IOExcep
     return MvtReader.loadMvt(mvtTile, geometryFactory, new TagKeyValueMapConverter(true, ID_KEY));
   }
 
+  /* Use the java port of the vector-tile-js library created by Dane Springmeyer.
+   * To get realistic number and have a fair comparison in terms of the decoding performance,
+   * the geometries of the features have to be decoded.
+   * */
+  public static Map<String, VectorTileLayer> decodeMvtMapbox(byte[] mvtTile) throws IOException {
+    Pbf pbf = new Pbf(mvtTile);
+    VectorTile vectorTile = new VectorTile(pbf, pbf.length);
+    for (var layer : vectorTile.layers.values()) {
+      for (int i = 0; i < layer.length; i++) {
+        var feature = layer.feature(i);
+        var geometry = feature.loadGeometry();
+      }
+    }
```

Comment:
Two thoughts on this:

1. Parity with MLT decoding in Java

See my comment above that I think this is doing significantly more decoding work than the MLT benchmark which currently appears to be calling `MltDecoder.decodeMlTileVectorized` without actually iterating the features/geometries.

3. Parity with JS

This is roughly equivalent to what I'm benchmarking against in JS. See here: https://github.com/maplibre/maplibre-tile-spec/blob/96852ba4577e849dfb29a3f14f3da7cf83d3f760/js/bench/utils.ts#L68-L88. The differences are:

A. I'm collecting the length of the top level geometry array as a cheap way to ensure that dead code elimination does not strike. I gather than jmh uses black hole techniques to avoid the JVM eliminating this code so that should not be needed? But maybe the java benchmarks should collect a count of something and return it just to be safe?
In JS I've also profiled to ensure that all these functions are being called as I would expect.

B. I'm sorting the layers first to workaround https://github.com/maplibre/maplibre-tile-spec/issues/190. I don't have evidence that parsing layers in different orders will result in different performance, but I could see it resulting in different memory allocations, so I think making sure the order is the same is a good idea to rule out any potential for variability due to that. Could you add sorting before the test?

Reply:
But that's also the point of the proposed MLT storage and in-memory format and why they are so tightly coupled and potentially show better decoding performance. Because you have to do less work and you can also work on compressed data